src/thing/get.test.ts

```diff
@@ -1139,7 +1157,100 @@ describe("getStringWithLocale", () => {
   });
 });
 
-describe("getStringsWithLocaleAll", () => {
+describe("getStringByLocale", () => {
+  function makeQuad(
+    subject: string,
+    predicate: string,
+    value: string,
+    locale?: string | NamedNode
+  ): Quad {
+    return DataFactory.quad(
+      DataFactory.namedNode(subject),
+      DataFactory.namedNode(predicate),
+      DataFactory.literal(value, locale)
+    );
+  }
+
+  it("returns all the string values for the given Predicate keyed by their locale", () => {
+    const thing = dataset()
+      .add(makeQuad(SUBJECT, PREDICATE, "value 1", "en"))
+      .add(makeQuad(SUBJECT, PREDICATE, "value 2", "en"))
+      .add(makeQuad(SUBJECT, PREDICATE, "value 3", "fr"))
+      .add(makeQuad(SUBJECT, PREDICATE, "value 4", "es"));
+
+    const thingWithLocaleStrings = Object.assign(thing, {
+      internal_url: SUBJECT,
+    });
+
+    expect(
+      Array.from(getStringByLocaleAll(thingWithLocaleStrings, PREDICATE))
+    ).toEqual([
+      ["en", ["value 1", "value 2"]],
+      ["fr", ["value 3"]],
+      ["es", ["value 4"]],
+    ]);
+  });
+
+  it("excludes non-language string values", () => {
```

Comment:
What happens for empty-string locales ? Are they ignored or added with an empty string key ?

Reply:
Great question. At first I thought we should handle this case specially, but in fact it seems an empty language tag is still perfectly valid, i.e. the literal is still of type `rdf:langString`. This made me curious about how that could be serialized in Turtle (since `<> rdfs:label "Some value"@ .` would definitely look like a syntax error!), but in fact the correct serialization is simply: `<> rdfs:label "Some value"^^rdf:langString .`. I demonstrated this in a CodeSandbox, but deploying to Vercel gave me an error (https://csb-izxlf-21qfb3kdk.vercel.app/).
Anyway I just extended the tests to check for this: commit 8c58c414